File: frontend/lib/models/scheduled_payment.dart
class ScheduledPayment {
  final int id;
  final String name;
  final double amount;
  final String frequency;
  final DateTime nextDue;
  final int? categoryId;
  final String? merchant;
  final bool isActive;

  ScheduledPayment({
    required this.id,
    required this.name,
    required this.merchant,
    required this.amount,
    required this.frequency,
    required this.nextDue,
    this.categoryId,
    required this.isActive,
  });

  bool get isExpense => amount < 0;

  String get formattedAmount {

    final abs = amount.abs().toStringAsFixed(2);
    return isExpense ? '-£$abs' : '+£$abs';

  }

  factory ScheduledPayment.fromJson(Map<String, dynamic> json) {
    return ScheduledPayment(
      id:         json['id'],
      name:       json['name'],
      amount:     (json['amount'] as num).toDouble(),
      merchant:   json['merchant'] as String? ?? '',
      frequency:  json['frequency'],
      nextDue:    DateTime.parse(json['next_due']),
      categoryId: json['category_id'] as int?,
      isActive:   json['is_active'] as bool? ?? true,
    );
  }
}

File: backend/schemas/scheduled_payment.py
from pydantic import BaseModel
from datetime import datetime

class ScheduledPaymentCreate(BaseModel):
    name: str
    amount: float
    frequency: str
    next_due: datetime
    category_id: int | None
    is_active: bool = True


class ScheduledPaymentUpdate(BaseModel):
    name: str | None
    amount: float | None
    frequency: str | None
    next_due: datetime | None
    category_id: int | None
    is_active: bool | None

    model_config = {"from_attributes": True}


class ScheduledPaymentSummary(BaseModel):
    id: int
    name: str
    amount: float
    frequency: str
    next_due: datetime
    category_id: int | None
    is_active: bool

    model_config = {"from_attributes": True}
